server.js

```diff
@@ -7967,6 +7967,70 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// Launchpad integration
+camp.route(/^\/launchpad\/ppa\/version\/([^/]+)\/([^/]+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  // The owner of the ppa
+  var user = match[1];
+  // The name of the ppa
+  var ppa = match[2];
+  // The package in the PPA to use
+  var name = match[3];
+  var format = match[4];
+
+  var badgeData = getBadgeData('launchpad', 'invalid');
+
+  // Make a request for packages in the PPA, returning only the published sources that match the the package
+  // Keep in mind that this will also return packages _starting_ with the package name
+  var apiUrl = 'https://api.launchpad.net/1.0/~' + user + '/+archive/' + ppa + '?ws.op=getPublishedSources&status=Published&source_name=' + name;
+  request(apiUrl, function(err, res, buffer) {
+    if (checkErrorResponse(badgeData, err, res)) {
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    try {
+      var data = JSON.parse(buffer);
+      var recentEntry = false;
```

Comment:
Lines 7994-8023 can be rewritten to:
```js
      const recentEntry = data.entries.filter(entry => entry.source_package_name == name).sort((a, b) => b.date_published.localeCompare(a.date_published))[0];
      if (recentEntry) {
        badgeData.text[1] = versionText(recentEntry.source_package_version);
        badgeData.colorscheme = versionColor(recentEntry.source_package_version);
      } else {
        badgeData.text[1] = 'no package';
        badgeData.colorscheme = 'red';
      }
      sendBadge(format, badgeData);
```
For loop is replaced by filter + sort. 
And 
```js
badgeData.colorscheme =versionText(recentEntry.source_package_version);
```
is replaced by:
```js
badgeData.colorscheme = versionColor(recentEntry.source_package_version);
```